### PR TITLE
Validate indices on zz merge

### PR DESCRIPTION
**Pull request by cdonati**

### CODE REVIEW

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/304/

File: AppDB/datastore_server.py [hunk 1]
```diff
@@ -1936,6 +1946,60 @@ def __fetch_entities_dict(self, refs, app_id):
 
     return clean_results
 
+  def __fetch_and_validate_entity_set(self, index_dict, limit, app_id,
+    direction):
+    """ Fetch all the valid entities as needed from references. Keep track of
+        invalid references.
```
Comment (menivaitsi) [inline AppDB/datastore_server.py]: 2 space indentation only please.

File: AppDB/datastore_server.py [hunk 2]
```diff
@@ -1936,6 +1946,60 @@ def __fetch_entities_dict(self, refs, app_id):
...
+
+    Args:
+      index_dict: A dictionary containing a list of index entries for each
+                  reference.
```
Comment (menivaitsi) [inline AppDB/datastore_server.py]: Same here.

**Comment by cdonati** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/309/

**Pull request merged**